server/public/model/channel.go

```diff
@@ -147,7 +147,6 @@ type ChannelPatch struct {
 	Header           *string            `json:"header"`
 	Purpose          *string            `json:"purpose"`
 	GroupConstrained *bool              `json:"group_constrained"`
-	Type             ChannelType        `json:"type"`
```

Comment:
Also, I noticed this was left in the previous PR even though we weren't using it.

I removed it since I want to make sure that no one accidentally ends up accidentally adding this because a bug like "Channel patch API can't update type field" is filed